ui/tests/integration/components/pki/pki-key-parameters-test.js

```diff
@@ -0,0 +1,98 @@
+
+  hooks.beforeEach(function () {
+    this.store = this.owner.lookup('service:store');
+    this.model = this.store.createRecord('pki/pki-role-engine');
+    this.model.backend = 'pki';
+  });
+
+  test('it should render the component and display the correct defaults', async function (assert) {
+    assert.expect(3);
+    await render(
+      hbs`
+      <div class="has-top-margin-xxl">
+        <PkiKeyParameters
+          @model={{this.model}}
+          @group="Key parameters"
+        />
+       </div>
+  `,
+      { owner: this.engine }
+    );
+    await click(SELECTORS.keyParams);
+    assert.dom(SELECTORS.keyType).hasValue('rsa');
+    assert.dom(SELECTORS.keyBits).hasValue('2048');
+    assert.dom(SELECTORS.signatureBits).hasValue('0');
+  });
+
+  test('it should set the model properties of key_type and key_bits when key_type changes', async function (assert) {
+    assert.expect(11);
+    await render(
+      hbs`
+      <div class="has-top-margin-xxl">
+        <PkiKeyParameters
+          @model={{this.model}}
+          @group="Key parameters"
+        />
+       </div>
+  `,
+      { owner: this.engine }
+    );
+    assert.deepEqual(this.model.keyType, 'rsa', 'sets the default value for key_type on the model.');
+    assert.deepEqual(this.model.keyBits, 2048, 'sets the default value for key_bits on the model.');
+    assert.deepEqual(this.model.signatureBits, 0, 'sets the default value for signature_bits on the model.');
+
+    await click(SELECTORS.keyParams);
+    await fillIn(SELECTORS.keyType, 'ec');
+    assert.strictEqual(this.model.keyType, 'ec', 'sets the new selected value for key_type on the model.');
+    assert.strictEqual(
+      this.model.keyBits,
+      256,
+      'dynamically sets the new selected value for key_bits on the model based on the selection of key_type.'
+    );
+
+    await fillIn(SELECTORS.keyType, 'ed25519');
+    assert.strictEqual(
+      this.model.keyType,
+      'ed25519',
+      'sets the new selected value for key_type on the model.'
+    );
+    assert.strictEqual(
+      this.model.keyBits,
+      0,
+      'dynamically sets the new selected value for key_bits on the model based on the selection of key_type.'
+    );
+
+    await fillIn(SELECTORS.keyType, 'ec');
+    await fillIn(SELECTORS.keyBits, 384);
+    assert.strictEqual(this.model.keyType, 'ec', 'sets the new selected value for key_type on the model.');
+    assert.strictEqual(
+      this.model.keyBits,
+      384,
+      'dynamically sets the new selected value for key_bits on the model based on the selection of key_type.'
+    );
+
+    await fillIn(SELECTORS.signatureBits, '384');
+    assert.strictEqual(
+      this.model.signatureBits,
+      384,
+      'sets the new selected value for signature_bits on the model.'
+    );
+
+    await fillIn(SELECTORS.signatureBits, '0');
+    assert.strictEqual(
+      this.model.signatureBits,
+      0,
+      'sets the default value for signature_bits on the model.'
```

Comment:
circling back to this flow because the default value comes from the defined default value on the model, but if you reselect this drop down, then it comes from the possibleValues on the model. I noticed a bug here (fixed) but figured it was a good idea to test for it.